### PR TITLE
Appointment

### LINKED ISSUE #4: As a medical doctor, I would like to schedule appointments with my patients so that I can talk to them one-on-one.
### **Description:**

As a medical doctor, I would like to schedule appointments with my patients. 

### **Steps:**

- [x] Add a Login page which redirects the user to the appropriate home page depending on user type.
- [x] Implement user profiles with associated user data. 
- [x] Create a "Book an Appointment" page where the doctor can select a patient and book an appointment with them. 
- [x] Send a notification of this appointment to the corresponding patient
- [x] Store appointment data on the database and fetch this data when requested. 

### **Acceptance Test Steps:**

- Log in using doctor credentials and land on the doctor home page.
- Click on the "Book an Appointment" tab to go to the Appointment booking page. 
- Enter in the required information and click "Book Appointment".
- Log in as the patient.
- Go to the notification page and see if there is a notification for an appointment. Verify appointment information. 
- The information on this page should be the same as the info you just entered in the "Update Status" page form.

### PR DESCRIPTION
**Pull request by p-carrier**
# General info
Allows the doctor to create appointments with a patient.

**Issue number**: #4 

**Task description**: 

 - Add create appointment button
 - Notify patient on appointment creation
 - Save appointment information
 - Fetch appointment information

# Testing
All test are working, make sure to install the new package in the frontend to review tests.

You should test to be able to:
- doctor should create appointment
- patient should receive appointment notification 
- doctor should fetch all its appointments

# Additional notes

**Note to reviewers**:

**To do before merging**:
- After merging, the us#4 will be done. Make sure to close #4

### CODE REVIEW

File: domain/doctor.js [hunk 1]
```diff
@@ -35,8 +43,32 @@ class Doctor {
     throw new Error(`${this.flagPatient.name} is not implemented.`);
   }
 
-  createAppointment(patient, date) {
-    throw new Error(`${this.createAppointment.name} is not implemented.`);
+  async createAppointment(patientId, appointmentInfo) {
+    // this.verifyDoctor();
```
Comment (AC159) [inline domain/doctor.js:47]: Why is the doctor verification commented?

File: domain/doctor.js [hunk 2]
```diff
@@ -35,8 +43,32 @@ class Doctor {
...
+      mainText: appointmentInfo.information,
+      subText: 'https//zoom.us/123456789',
+      timestamp: Date.now(),
+      userId: patientId,
+    };
+    await this.doctorRepository.insertNotification(notification);
+  }
+
+  async getAppointments() {
+    // this.verifyDoctor();
```
Comment (AC159) [inline domain/doctor.js:70]: Same here

File: repository/DoctorRepository.js [hunk 3]
```diff
@@ -31,6 +27,31 @@ class DoctorRepository {
         .next();
   }
 
+  // getPatientInfo(patientId) {
+  //   return this.mongo.db('test').collection('user')
+  //       .find({uid: patientId});
```
Comment (AC159) [inline repository/DoctorRepository.js:32]: If this is not used, it should be removed

Comment (p-carrier) [reply repository/DoctorRepository.js:32]: You are right. I'll remove it

File: routes/doctor.js [hunk 4]
```diff
@@ -17,6 +17,32 @@ router.get('/:doctorId/patientArray', async function(req, res) {
   }
 });
 
+router.post('/:doctorId/appointment', async (req, res) => {
```
Comment (AC159) [inline routes/doctor.js:20]: minor detail, but maybe we can rename this route to 'create-appointment' to make it clearer

**Review by AC159 — CHANGES_REQUESTED** [review]
See comments

**Review by DrakeHunterK7 — APPROVED** [review]
Tests are passing, features are working, life is good.

**Review by AC159 — APPROVED** [review]
Good to go

**Pull request merged**